data/pools/transactionPool.go

```diff
@@ -132,18 +132,28 @@ type poolAsmResults struct {
 	// roundStartedEvaluating is the round which we were attempted to evaluate last. It's a good measure for
 	// which round we started evaluating, but not a measure to whether the evaluation is complete.
 	roundStartedEvaluating basics.Round
+	// assemblyComplete is *not* protected via the pool.assemblyMu lock and should be accessed only from the OnNewBlock goroutine.
+	// it's equivilent to the "ok" variable, and used for avoiding taking the lock.
+	assemblyComplete bool
```

Comment:
Maybe call this assemblyCompletedOrAbandoned to reflect that fact that the block might not be really assembled when this is set to true.

Reply:
I'm fine with changing this to assemblyCompletedOrAbandoned, although I don't think it adds much information. ( I'll wait for the other reviewers to comment here before making this change )